### PR TITLE
Adding AVA testing framework

### PR DESCRIPTION
**Pull request by adrichey**
Added AVA as the testing framework, nbc for code coverage reports, and ninon for testing functions using says and stubs. Client tests were also implemented.

### CODE REVIEW

File: test/Client.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+const Domain = require("../src/Domain");
+const Link = require("../src/Link");
+const Script = require("../src/Script");
+const Tag = require("../src/Tag");
+const Client = require("../src/Client");
+
+/**
+ * constructor
+ */
+test('Client constructor sets and stores the API key', t => {
```
[the remark this answers is not in the document]
Comment (adrichey) [reply test/Client.js]: Done! Updated and pushed.

[the remark this answers is not in the document]
Comment (adrichey) [reply test/Client.js]: I plan on writing more tests this week. Last week was swamped at work, so fun projects took a back seat.

[the remark this answers is not in the document]
Comment (ifvictr) [reply test/Client.js]: Could you also change it so that you're not using `import`?

[the remark this answers is not in the document]
Comment (adrichey) [reply test/Client.js]: Done.

**Comment by ifvictr** [comment]
@adrichey Merged! :tada:

**Pull request merged**